dts/bindings/power/state.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2020, Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Properties for power management state
+
+compatible: "pm-state"
+
+properties:
+    pm-state:
+        type: string
+        required: true
+        description: indicates a power state
+        enum:
+            - "PM_STATE_ACTIVE"
```

Comment:
my two cents, I don't think `_ENUM_TOKEN` solve the main problem that is having this information duplicated in both domains (C and DTS). One still need to follow a name convention and we need to declare these values in dts bindings. That is said, the thing that `ENUM_TOKEN` gives to me is that I don't need to declare the states in the correct order in the binding.

I'm totally ok in using `ENUM_TOKEN` here when it gets merged, I just don't think that it should gate this pr.

Reply:
It kinda has to, unless you have another way to set this up so the enum values don't include the `PM_STATE_` prefix.  Because that should not be there, and we shouldn't merge with bindings that we intend to change.

I think we should be able to get this in within a week, there're only a couple of bikeshed color choices relevant to enum token support.

Reply:
why can't it be changed later ? I mean, if the problem is changing the binding later, I can remove the `PM_STATE` prefix.
I've more code depending in these basic changes, the whole power subsystem is still using old states and getting info from Kconfig. I want to move on in these changes .... this `ENUM_TOKEN` change will be quite small and only in the macro to get it and should not impact the rest of the changes.

Reply:
Yes, please remove the `PM_STATE_` prefix from the binding `pm-state` property enumeration values.  If you can make that work without enum token support in devicetree or removing the `PM_STATE_` prefix from the pm_state C enum values, that'd be fine.

Reply:
> [ @mbolivar-nordic ] would you have time to work on `_ENUM_TOKEN` or should I take it on?

As discussed, I'm on it

Reply:
thank you